feat(agent): Honor collect_span_events server config

misiekn opened the pull request:
Agent will honor server side configuration and will not collect span 
events if this feature is disabled in the configuration received from 
the server. Server side configuration takes precedence over application's
local configuration. However server side configuration can only be used 
to turn off collecting span events.

Closes #443

zsistla commented:
Thanks for getting this up! For completeness, should we toggle collect_span_events on in the initialization of other tests?  For instance: https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/tests/test_txn.c#L5250

zsistla commented:
Looks great overall.

mfulb reviewed (approved):
Nice contribution Michal.

misiekn commented:
> Thanks for getting this up! For completeness, should we toggle collect_span_events on in the initialization of other tests? For instance: https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/tests/test_txn.c#L5250

I do not think this is necessary because the only change in the behavior of the agent is when `connect_reply` has `collect_span_events` attribute with the value of 0 (false). This behavior is tested by the newly added test [axiom/tests/test_txn.c#L1780](https://github.com/misiekn/newrelic-php-agent/blob/0a2a5d1ccdc66af03a52f42d14d88bcf101d4e82/axiom/tests/test_txn.c#L1780). Any other value is ignored by agent and this behavior is tested by all existing tests in [test_begin](https://github.com/misiekn/newrelic-php-agent/blob/0a2a5d1ccdc66af03a52f42d14d88bcf101d4e82/axiom/tests/test_txn.c#L1629) function, which compare results of `nr_txn_enforce_security_settings` (called by `nr_txn_begin`) with expected value `correct.span_events_enabled` declared [here](https://github.com/misiekn/newrelic-php-agent/blob/0a2a5d1ccdc66af03a52f42d14d88bcf101d4e82/axiom/tests/test_txn.c#L1698).

On the other hand tests in [test_sql_recording_level_lasp](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/tests/test_txn.c#L5250) require `connect_reply` to have certain lasp attributes.

Hopefully this addresses your concerns.
